### PR TITLE
Add Argo Rollouts Analysis demo with Pixie metrics.

### PR DESCRIPTION
**Pull request by htroisi**
To Do:
- [x] Get Michelle to build & push image to `gcr.io/pixie-oss/pixie-dev/demo/argo-rollouts-demo`.
- [x] Update `px-metrics.yaml`

Signed-off-by: Hannah Troisi <htroisi@pixielabs.ai>

### CODE REVIEW

File: argo-rollouts-demo/pixie-demo.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  labels:
+    name: px-metrics-service
+  namespace: px-metrics
+spec:
+  type: LoadBalancer
+  ports:
+    - port: 80
+      targetPort: 8080
+  selector:
+    name: px-metrics-service
```
Comment (nserrino) [inline argo-rollouts-demo/pixie-demo.yaml:38]: add newline

File: argo-rollouts-demo/svc_error_rate.pxl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+df = df.groupby('service').agg(
+    http_req_count_in=('latency', px.count),
+    http_error_count_in=('failure', px.sum),
+)
+
+df.http_error_rate_in = px.Percent(
+        px.select(df.http_req_count_in != 0, df.http_error_count_in / df.http_req_count_in, 0.0))
+
+df = svc_list.merge(df, how='left', left_on='service', right_on='service', suffixes=['', '_x'])
+px.display(df[['service', 'http_error_rate_in']], 'pod_stats')
```
Comment (nserrino) [inline argo-rollouts-demo/svc_error_rate.pxl:34]: add newline

File: argo-rollouts-demo/pixie-server.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	b, err := ioutil.ReadFile("svc_error_rate.pxl")
+	if err != nil {
+		panic(err)
+	}
+	pxlScript := string(b)
+
+	// Compute metrics
+	ctx := context.Background()
+	p.computeMetrics(ctx, pxlScript)
+	//log.Println("New Pixie metrics: ", p.podRequestsPerS)
```
Comment (nserrino) [inline argo-rollouts-demo/pixie-server.go:48]: clean up

File: argo-rollouts-demo/README.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+kubectl get svc canary-demo-preview
+```
+
+4. Navigate to the External IP in your browser to access the demo application front-end. Leave this open.
+
+<br>
+<img src=".readme_assets/front-end.png"  width="550" align="right">
+<br>
+
+Each of these colorful squares is a request the browser is making to the backend.
```
Comment (aimichelle) [inline argo-rollouts-demo/README.md:85]: Got a little confused which backend was being referenced here. Maybe to be really clear, change it to "Each of these colorful squares represents a request made by the browser to the demo application backend".

File: argo-rollouts-demo/README.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+The backend responds with a color that indicates which version of the app it is. In this case we're getting blue boxes for the application image with the `blue` tag set in the [Rollout](https://github.com/pixie-io/pixie-demos/tree/main/argo-rollouts-demo/canary/rollout-with-analysis.yaml) yaml.
+
+The bar chart at the bottom represents the percentage of requests that were handled by the different backends (stable, canary). We will see this in action in a minute.
```
Comment (aimichelle) [inline argo-rollouts-demo/README.md:89]: "The bar chart at the bottom represents the percentage of requests that were handled by the different backends (stable, canary). **Currently you should see all requests are handled by the "stable" backend.**  We will see this in action in a minute."

File: argo-rollouts-demo/README.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+
+<br>
+<img src=".readme_assets/successful-rollout.png" width="550" align="right">
+<br>
+
+On our first step, we direct 50% of traffic to the `blue` release and 50% to `yellow` release.
+
+Argo Rollouts splits traffic between versions by creating a new replica set that uses the same service object and the service will still split the traffic evenly across pods (new and old). In other words, controlling the number of pods controls the traffic percentage.
+
+Our analysis is measuring the HTTP error rate for the canary pods every 30 seconds.
```
Comment (aimichelle) [inline argo-rollouts-demo/README.md:117]: Read through the whole README, and it's not clear where Pixie plays into this... I think we need to add a point here to make it very clear that the Pixie metrics server is what is supplying this HTTP error rate.

File: argo-rollouts-demo/px-metrics.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      labels:
+        name: px-metrics
+        plane: control
+    spec:
+      containers:
+        - name: app
+          image: gcr.io/pixie-oss/pixie-dev/demo/argo-rollouts-demo:latest
+          env:
+          - name: PX_CLOUD_ADDR
+            value: withpixie.ai:443
```
Comment (aimichelle) [inline argo-rollouts-demo/px-metrics.yaml:24]: We can't do this, unfortunately. We may need to point to self hosted cloud instructions and tell them to fill this in.

Comment (htroisi) [reply argo-rollouts-demo/px-metrics.yaml:24]: I will update this once the `px get cloudAddr` command lands.

File: argo-rollouts-demo/px_metrics_server/pixie-metric-server.go [hunk 8]
```diff
@@ -0,0 +1,211 @@
+/*
```
Comment (aimichelle) [inline argo-rollouts-demo/px_metrics_server/pixie-metric-server.go:1]: This is the wrong license. We should be using the Apache 2.0

**Review by aimichelle — APPROVED** [review]

**Pull request merged**